### PR TITLE
Add support for `AS table_alias` syntax for inserts in Postgres

### PR DESCRIPTION
**Pull request by CDThomas**
Adds support for using table aliases in inserts.

### CODE REVIEW

**Comment by CDThomas** [comment]
Could use some more tests and possibly some refactoring, but this should do it.

File: src/parser/mod.rs [hunk 1]
```diff
@@ -7105,6 +7105,15 @@ impl<'a> Parser<'a> {
             // Hive lets you put table here regardless
             let table = self.parse_keyword(Keyword::TABLE);
             let table_name = self.parse_object_name()?;
+
+            let is_postgresql = dialect_of!(self is PostgreSqlDialect);
```
Comment (coderdan) [inline src/parser/mod.rs:7109]: According [to the docs](https://github.com/cipherstash/sqlparser-rs/tree/main#new-syntax), dialect specific syntax needs to be handled for both the dialect in question and the `GenericDialect`.

Comment (CDThomas) [reply src/parser/mod.rs:7109]: I think that only applies when you need to make changes to the dialect itself. I don't think that matters here, but I'll see what the maintainer says when I raise it upstream.

I still need to check other dialects as well to verify that this needs to be dialect specific.

Comment (coderdan) [reply src/parser/mod.rs:7109]: Isn't this doing exactly that? Changing the dialect?

**Comment by CDThomas** [comment]
Closing since this has been fixed upstream: https://github.com/sqlparser-rs/sqlparser-rs/pull/1084

**Pull request closed without merging**